test/test_nn.py

```diff
@@ -161,6 +161,14 @@ def fn(a, b=None, *, c=None):
             td = TensorDict({"1": torch.ones(1)}, [])
         assert (module(td)["a"] == 2).all()
 
+    def test_tuple_input_keys(self):
+        in_keys = {"1": "first", "2": ("second", "third")}
+        fn = lambda x, y, z: x + y + z
+        module = TensorDictModule(fn, in_keys=in_keys, out_keys=["out"])
+
+        kword_in_keys = sorted(zip(module._kwargs, module.in_keys))
+        assert kword_in_keys == [("first", "1"), ("second", "2"), ("third", "2")]
```

Comment:
I don't really get this example:
the modue is not run, what would the input tensordict look like?
Could the example be a bit more extensive?

Reply:
Thanks, it's a good point - I updated that test. I also added another test that tests the integration of TensorDictModule within a ProbabilisticTensorDictSequential.

Reply:
@vmoens Thanks for your feedback.
Could you clarify how you would see the change with `ProbabilisticTensorDictModule`? It seems to me that this feature will only impact `TensorDictModule`. `ProbabilisticTensorDictModule` inherits from `TensorDictModuleBase` and is not affected. Also I don't see the use of implementing my suggested feature in `ProbabilisticTensorDictModule` - its input have to be unique as far I see it.
Please let me know your thoughts

Reply:
Sure
What if I want to do something along the line of 
```python
ProbabilisticTensorDictModule(
   in_keys={"hidden": ("concentration", "rate")},
   out_keys=["other"],
   distribution_class=torch.distributions.Gamma
)
```
with the [`gamma`](https://pytorch.org/docs/stable/distributions.html#torch.distributions.gamma.Gamma) distribution taking params `concentration` and `rate`.

As per the utility of doing such thing - I think it's up to the users. To me as developer is more a matter of keeping the API as uniform as we can. You could for instance decide to build 2 different Normal distributions with a `CompositeDistribution` class that share the same scale?

Reply:
Thanks for the explanations @vmoens . 
Sorry, I am still a bit confused - the API documentation states that ` If in_keys is a dictionary, the keys are the keys of the distribution and the values are the keys in the tensordict that will get match to the corresponding distribution keys.`. In your example, it looks to me that you should pass the in_keys dictionary in the reverse order. Passing the same tensordict key to 2 different distribution parameters seems to be already supported, for instance via:

```python
ProbabilisticTensorDictModule(
   in_keys={"concentration": "value_in_input_td", "rate": "value_in_input_td" },
   out_keys=["other"],
   distribution_class=torch.distributions.Gamma
)
```

We could change the API to match your example but I suppose that it would be a breaking API change, wouldn't it?

Reply:
Yeah i just realized that, so we are asking dicts that are flipped between TensorDictModule and ProbabilisticTensorDictModule?
That ain't good!
If we flip the dicts in TensorDictModule (to match Proba), maybe that would solve your problem altogether?

Reply:
Absolutely, that would change the problem altogether! And it's a relatively change to make (docs and tests I assume)
However, that would be a breaking change for users already using dictionary with TensorDictModule. How would you like to proceed?

Reply:
Yep that's BC breaking but it's for the best
Here is how I would procede in this case: what we want is that people passing `in_keys` now with a dict don't see things breaking but they get a warning, and we also want to make your life easy if you're using the new API. 
So I'd suggest to add a `out_to_in_map: bool | None = None` keyword arg in the module that says:
- if `None`, the current default is used but if `in_keys` is a dict, then a deprec warning is raised
- If `False` we use the current behaviour `input-key: output-key`, no warning is raised.
- If `True`, we use the new behaviour.

The warning must be explicit and tell the user that the default will change 2 versions from now (next is 0.7 so it will be in 0.9). To silence the warning pass `out_to_in_map=False` or `True`.

Does that makes sense? Should I take care of it?

Reply:
Yes, thanks, I like this approach, it's safe. I can push the change today and add some tests for the 3 cases (check the warning is raised etc)